File: Algorithms/513.FindBottomLeftTreeValue.py
class Solution(object):
    def findBottomLeftValue(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        cur_level, next_level, last_level = [], [], []
        cur_level.append(root)
        value = root.val
        while cur_level:
            node = cur_level.pop(0)
            if node.left:
                next_level.append(node.left);
            if node.right:
                next_level.append(node.right);
            if not cur_level:
                cur_level = next_level;
                if cur_level:
                    value = cur_level[0].val
                next_level = []
        return value

File: Algorithms/java/513.FindBottomLeftTreeValue.java
class Solution {
    public int findBottomLeftValue(TreeNode root) {
        List<TreeNode> cur_level = new ArrayList<>();
        List<TreeNode> next_level = new ArrayList<>();
        cur_level.add(root);
        int value = root.val;
        while(cur_level.size() > 0) {
            TreeNode node = cur_level.get(0);
            cur_level.remove(0);
            if (node.left != null) next_level.add(node.left);
            if (node.right != null) next_level.add(node.right);
            if (cur_level.size() == 0) {
                cur_level = next_level;
                if (cur_level.size() != 0)
                    value = cur_level.get(0).val;
                next_level = new ArrayList<>();
            }
        }
        return value;
    }
}
